platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/cache/VirtualNodeCache.java

```diff
@@ -148,32 +148,38 @@ private static final class ClassVersion {
      */
     public static final Hash NULL_HASH = new Hash();
 
-    /**
-     * Since {@code com.swirlds.platform.Browser} populates settings, and it is loaded before any
-     * application classes that might instantiate a data source, the {@link ConfigurationHolder}
-     * holder will have been configured by the time this static initializer runs.
-     */
-    private static final VirtualMapConfig config = ConfigurationHolder.getConfigData(VirtualMapConfig.class);
-
-    /**
-     * This thread pool contains the threads that purge unneeded key/mutation list pairs from the indexes.
-     * The "syncCleaningPool" property is used for testing.
-     */
-    private static final Executor CLEANING_POOL = Boolean.getBoolean("syncCleaningPool")
-            ? Runnable::run
-            : new ThreadPoolExecutor(
-                    config.getNumCleanerThreads(),
-                    config.getNumCleanerThreads(),
-                    60L,
-                    TimeUnit.SECONDS,
-                    new LinkedBlockingQueue<>(),
-                    new ThreadConfiguration(getStaticThreadManager())
-                            .setThreadGroup(new ThreadGroup("virtual-cache-cleaners"))
-                            .setComponent("virtual-map")
-                            .setThreadName("cache-cleaner")
-                            .setExceptionHandler((t, ex) -> logger.error(
-                                    EXCEPTION.getMarker(), "Failed to purge unneeded key/mutationList pairs", ex))
-                            .buildFactory());
+    private static final AtomicReference<Executor> cleaningPool = new AtomicReference<>();
+
+    private static Executor getCleaningPool() {
+        Executor exec = cleaningPool.get();
+        if (exec == null) {
```

Comment:
Minor: again same as above. I'd definitely recommend to create an utility holder class at this point as suggested above.